internal/envoy/cluster.go

```diff
@@ -100,6 +100,38 @@ func Cluster(c *dag.Cluster) *v2.Cluster {
 	return cluster
 }
 
+// ExtensionCluster builds a v2.Cluster struct for the given extension service.
+func ExtensionCluster(ext *dag.ExtensionService) *v2.Cluster {
+	c := Cluster(&dag.Cluster{
+		Upstream:           &dag.Service{},
```

Comment:
Yeh, the weighting happens when the `.Upstream` field is processed by the DAG and that generates a `ServiceCluster`. Here, I just want to build a `v2.Cluster that supports the common fields without duplicating a bunch of code.

Reply:
It's feels bad to require this setup here and assume it happens later on. Also where does weight get added if not here? Shouldn't that weight come from the CRD spec and be applied into this field?

Reply:
The weight is all in the Upstream, which is a ServiceCluster. That's handled directly by the endpoints translator.

Reply:
Maybe this needs a `NewCluster` method or something to hide these impl details?

Reply:
> Maybe this needs a `NewCluster` method or something to hide these impl details?

I'm not super keen on that since we don't have that pattern in general in the `dag` package. This use is kinda hack just to avoid code duplication. I'd prefer to clean this part up in the future.

Reply:
I just removed the hackery and built the extension service cluster directly. It's easier to understand and I should have just done that in the first place :)

Reply:
@stevesloka I'm going to resolve this and merge, but let's check in tomorrow and I'm happy to make changes in a new PR if have more feedback.